reason_4.0/lib/core/classes/api/feed/feed.php

```diff
@@ -110,7 +110,11 @@ class ReasonFeedAPI extends CarlUtilAPI
 				if ($json = $model->get())
 				{
 					$this->set_content($json);
-				}	
+					// pull out the status and set it on the response
+					$decoded_json = json_decode($json, true);
```

Comment:
An alternate approach would be to add an additional method to the model, like `get_response_code()`, which would remove the need to parse the output and reduce the likelihood of a namespace collision (e.g. some json format that uses a "status" key).

Then the code on this side would look something like:
```php
$this->set_http_response_code($model->get_response_code());
```